### PR TITLE
Updating gallery layout to use Robs distribution logic from v2

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/cards/src/GalleryLayout.tsx [hunk 1]
```diff
@@ -116,16 +173,35 @@ export class GalleryLayout<T> extends BaseLayout<T> {
...
-    let partition = linearPartition(ratios, rows);
+    let rows = Math.max(1, Math.ceil(totalWidth / availableWidth));
+    // if the available width can't hold two items, then every item will get its own row
+    // this leads to a faster run through linear partition and more dependable output for small row widths
+    if (availableWidth <= (this.minItemSize.width * 2) + (this.itemPadding * 2)) {
+      rows = this.collection.size;
+    }
+
+    let weightedRatios = ratios.map(ratio => ratio < this.threshold ? ratio + (0.5 * (1 / ratio)) : ratio);
+    let partition = linearPartition(weightedRatios, rows);
```
Comment (LFDanLu) [inline packages/@react-spectrum/cards/src/GalleryLayout.tsx:204]: This partitioning doesn't quite respect the minItemSize.height provided, will need to dig into how the linearPartition works. May require a refactor of these calculations but that can be done after this alpha

**Comment by snowystinger** [comment]
I wonder, should min-width only apply when there is body content (title/actions/description/footer) so that if it's just the image, they look nice and flush? check out google photos gallery view. sometimes the checkbox extends outside the image

**Comment by LFDanLu** [comment]
@snowystinger hm, a bit tricky to do with the currently implementation since the layout doesn't know anything about the card contents but perhaps something to explore with a refactor to the layout as a whole. I also wonder about how it would look if the user mixes cards with content and cards that only have the image

File: packages/@react-spectrum/cards/src/GalleryLayout.tsx [hunk 2]
```diff
@@ -117,16 +174,35 @@ export class GalleryLayout<T> extends BaseLayout<T> {
     // Compute aspect ratios for all of the items, and the total width if all items were on in a single row.
     let ratios = [];
     let totalWidth = 0;
+    let minRatio = this.minItemSize.width / this.minItemSize.height;
```
Comment (LFDanLu) [inline packages/@react-spectrum/cards/src/GalleryLayout.tsx:177]: Note that this still has a bit of an issue where it is possible for the item's height for extremely tall images to be a bit taller than its row peers:
image

Comment (snowystinger) [reply packages/@react-spectrum/cards/src/GalleryLayout.tsx:177]: well that doesn't seem good, lets sit down and go through the math together sometime

**Pull request merged**